### PR TITLE
Update fault codes to better reflect config model.

### LINKED ISSUE #378: Update fault codes to better align with configuration model.
The existing fault codes in fault/fault_codes.go do not reflect the configuration model well. We should update the codes to better match our configuration, so that they'll make more sense to those who consume them.

### PR DESCRIPTION
**Pull request by emschwar**
* Added a new code to better distinguish "api key not found" from
  "api key not present."

Closes #378

### CODE REVIEW

**Review by kansalarchit — COMMENTED** [review]
I thought we were going to re-use the existing Apigee MP fault codes, is that not the case anymore ?

File: fault/fault_codes.go [hunk 1]
```diff
@@ -15,15 +15,16 @@
...
-	JwtAudienceMismatch           = "steps.jwt.JwtAudienceMismatch"
+	UnknownAPIProxy               = "environment.ApiProxyNotFound"
+	UnknownKeyManagementException = "consumerAuthorization.InternalError"
+	AuthorizationCodeNotFound     = "consumerAuthorization.RequestMissingKey"
+	InvalidAuthorizationCode      = "consumerAuthorization.UnknownKey"
+	InternalError                 = "apiProxy.InternalError"
+	NoApiProductMatchFound        = "consumerAuthorization.NoApiProductMatchFound"
+	OperationQuotaExceeded        = "consumerAuthorization.quota.LimitExceeded"
+	InternalQuotaError            = "consumerAuthorization.quota.InternalError"
+	JwtUnknownException           = "authentication.jwt.UnknownException"
```
Comment (KyleWiese) [inline fault/fault_codes.go:26]: authentication.jwt.InternalError

**Review by KyleWiese — CHANGES_REQUESTED** [review]

File: fault/fault_codes.go (hunk 1 above)
Comment (emschwar) [reply fault/fault_codes.go:26]: Done.

**Comment by emschwar** [comment]
> I thought we were going to re-use the existing Apigee MP fault codes, is that not the case anymore ?

We've decided those codes would be more confusing than helpful at this point.

**Review by KyleWiese — DISMISSED** [review]

**Comment by theganyo** [comment]
Should we be checking that our faults correctly get added to the request? For example, the headers here: https://github.com/apigee/apigee-remote-service-envoy/blob/master/server/authorization_test.go#L343

**Review by kansalarchit — APPROVED** [review]

**Comment by emschwar** [comment]
@theganyo I added some tests for the fault code headers; PTAL.

**Review by theganyo — APPROVED** [review]

**Comment by theganyo** [comment]
Thanks!

**Review by kansalarchit — DISMISSED** [review]

**Review by theganyo — APPROVED** [review]

**Pull request merged**